packages/integration-tests/src/retry.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Executes an asynchronous operation with retry logic.
+ * This function attempts to execute the provided callable function multiple times
+ * based on the specified retry conditions. It's useful for handling transient
+ * errors or temporary service unavailability.
+ */
+export const runWithRetry = async <T>(
+  callable: () => Promise<T>,
+  retryPredicate: (error: Error) => boolean,
+  maxAttempts = 3,
+  timeout?: number
+): Promise<T> => {
+  const startTime = Date.now();
+  const collectedErrors: Error[] = [];
+
+  for (let attempt = 1; attempt <= maxAttempts; attempt++) {
+    try {
+      if (timeout && Date.now() - startTime >= timeout) {
+        throw new Error(`Operation timed out after ${timeout}ms`);
+      }
```

Comment:
I'd remove timeouts from here for now.
We configure timeouts in heath_checks.yml, so this is redundant.